### PR TITLE
add scorecard algos extensions

### PR DESCRIPTION
**Pull request by ckeys**

### CODE REVIEW

**Comment by allwefantasy** [comment]
I do not seen where the  three ETs  are registered.

**Review by allwefantasy — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -80,7 +80,7 @@
         <serviceframework.version>2.0.5</serviceframework.version>
         <scope>provided</scope>
         <hadoop-client-version>2.6.5</hadoop-client-version>
-        <mlsql.version>2.2.0-SNAPSHOT</mlsql.version>
+        <mlsql.version>2.3.0-SNAPSHOT</mlsql.version>
```
Comment (allwefantasy) [inline pom.xml:83]: You should change the template pom file in .repo directory. The deploy script will evaluate the template pom file in .repo to replace the original pom file.

File: pom.xml [hunk 2]
```diff
@@ -108,6 +108,13 @@
             <scope>${scope}</scope>
         </dependency>
 
+        <dependency>
```
Comment (allwefantasy) [inline pom.xml:111]: Can you tell me why we need to declare this dependency?

**Review by allwefantasy — CHANGES_REQUESTED** [review]

File: mlsql-mllib/src/main/java/tech/mlsql/plugins/mllib/ets/fintech/scorecard/SQLBinning.scala [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+    val context = ScriptSQLExec.contextGetOrForTest()
+    context.execListener.getTableAuth match {
+      case Some(tableAuth) =>
+        tableAuth.auth(List(vtable))
+      case None =>
+        List(TableAuthResult(granted = true, ""))
+    }
+  }
+
+}
```
Comment (allwefantasy) [inline mlsql-mllib/src/main/java/tech/mlsql/plugins/mllib/ets/fintech/scorecard/SQLBinning.scala:505]: We should define all params in ET way to support ET self-inspecting, aka. code auto complete.

File: mlsql-mllib/src/main/java/tech/mlsql/plugins/mllib/ets/fintech/scorecard/SQLScoreCard.scala [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+      TableType.SYSTEM)
+
+    val context = ScriptSQLExec.contextGetOrForTest()
+    context.execListener.getTableAuth match {
+      case Some(tableAuth) =>
+        tableAuth.auth(List(vtable))
+      case None =>
+        List(TableAuthResult(granted = true, ""))
+    }
+  }
```
Comment (allwefantasy) [inline mlsql-mllib/src/main/java/tech/mlsql/plugins/mllib/ets/fintech/scorecard/SQLScoreCard.scala:264]: Also this class is lack of  ET parameter defining.

**Review by allwefantasy — CHANGES_REQUESTED** [review]

File: mlsql-mllib/src/main/java/tech/mlsql/plugins/mllib/ets/fintech/scorecard/SQLBinning.scala (hunk 3 above)
Comment (ckeys) [reply mlsql-mllib/src/main/java/tech/mlsql/plugins/mllib/ets/fintech/scorecard/SQLBinning.scala:633]: done

**Pull request merged**